File: 2D_amber/Assets/Scripts/JS/Behaviour/ShotBehaviour.js
#pragma strict

class ShotBehaviour extends StateMachineBehaviour{
  function OnStateUpdate(anim:Animator, info:AnimatorStateInfo, id:int){
    anim.GetComponentInParent.<Rigidbody2D>().drag = 20 ;
  }
  function OnStateExit(anim:Animator, info:AnimatorStateInfo, id:int){
    anim.GetComponentInParent.<Rigidbody2D>().drag = 0 ;
  }
  
}

File: 2D_amber/Assets/Scripts/CSharp/Behaviour/ShotBehaviour.cs
using UnityEngine;
using System.Collections;

public class ShotBehaviour : StateMachineBehaviour {
	
	void OnStateUpdate(Animator anim, AnimatorStateInfo info, int id){
    	anim.GetComponentInParent<Rigidbody2D>().drag = 20 ;
  	}
  	void OnStateExit(Animator anim, AnimatorStateInfo info, int id){
    	anim.GetComponentInParent<Rigidbody2D>().drag = 0 ;
  	}
}
